asana/README.md

```diff
@@ -7,10 +7,98 @@ A connector to integrate Asana, the project management tool, to Cohere. Featurin
 The Asana connector only offers full-text search for Task objects.
 
 ## Configuration
+The Asana connector provides two authentication methods: [Personal Access Token](https://developers.asana.com/docs/personal-access-token) and [OAuth 2.0](https://developers.asana.com/docs/oauth)
 
-To use this connector, head to your Asana Developer Console. From here, you can create a new Personal Access Token. Use this value for the `ASANA_ACCESS_TOKEN` environment variable. Then, head to your Admin Console. The URL at the top of your web browser should look like `https://app.asana.com/admin/<workspace_gid>`, grab the `<workspace_gid>` value and use it for `ASANA_WORKSPACE_GID`.
+If you are using a personal access token, you need to set the following environment variables:
+```
+ASANA_AUTH_TYPE=access_token
+```
+and head to your Asana Developer Console. From here, you can create a new Personal Access Token. Use this value for the `ASANA_ACCESS_TOKEN` environment variable.
+Additionally, to safeguard the connector from abuse, set the `ASANA_CONNECTOR_API_KEY` environment variable to a secure value for bearer token authentication. 
+Do not set this variable if you are using OAuth 2.0 authentication.
+
+If you are using OAuth 2.0, you need to set the following environment variables:
+```
+ASANA_AUTH_TYPE=oauth
+```
+and head to your Asana Developer Console. From here, you can create a new OAuth App. Use the Client ID and Client Secret values
+to register the connector with Cohere's API using Oauth 2.0.
+Also, you need to set the Redirect URL to `https://api.cohere.com/v1/connectors/oauth/token` on the Oauth App settings page.
+Here is an example of how to register the connector with Cohere's API using Oauth 2.0:
+```bash
+ curl  -X POST \
+   'https://api.cohere.ai/v1/connectors' \
+   --header 'Accept: */*' \
+   --header 'Authorization: Bearer {COHERE-API-KEY}' \
+   --header 'Content-Type: application/json' \
+   --data-raw '{
+   "name": "Asana with OAuth",
+   "url": "{YOUR_CONNECTOR-URL}",
+   "oauth": {
+     "client_id": "{Your Asana App CLIENT-ID}",
+     "client_secret": "{Your Asana App SECRET}",
+     "authorize_url": "https://app.asana.com/-/oauth_authorize",
+     "token_url": "https://app.asana.com/-/oauth_token",
+   }
+ }'
+```
+
+Then, head to your Admin Console. The URL at the top of your web browser should look like `https://app.asana.com/admin/<workspace_gid>`, 
+grab the `<workspace_gid>` value and use it for `ASANA_WORKSPACE_GID`.
+
+## Optional Configuration
+```
+ASANA_TASK_PROPERTIES
+```
+This variable may contain a comma-separated list of task properties (e.g. ["name","notes"]).
+By default, the following task properties are returned:
```

Comment:
Can you link the docs for the task properties here

Reply:
Thanks. It's done